components/markdown/markdown.jsx

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import messageHtmlToComponent from 'utils/message_html_to_component';
+import * as TextFormatting from 'utils/text_formatting.jsx';
+
+export default class Markdown extends React.PureComponent {
```

Comment:
Would be great to have tests both for `CustomUrlSchemesSetting` and `Markdown` components.

Reply:
I added some brief ones for the component and then some tests for the new option to the TextFormatting tests